mooncaker/external_tools/mooncaker_bot.py

```diff
@@ -0,0 +1,149 @@
+from functools import partial
+from telegram import Update, ForceReply, Sticker, InlineKeyboardButton, InlineKeyboardMarkup
+from telegram.ext import Updater, CommandHandler, MessageHandler, Filters, CallbackContext, CallbackQueryHandler, \
+    ConversationHandler
+from os import remove
+from mooncaker.external_tools.db_interactor import Database
+
+
+class MooncakerBot:
+    SAD_ZOE = 'CAACAgIAAxkBAAECkWNg7Fo-k6squNhgN_2bLL2X3F6DhQACWwADsiFfFVNQKDUCqm9IIAQ'
+    MEGUMIN_THUMB_UP = 'CAACAgUAAxkBAAECkXNg7GRAgV-Bw0QddeuTXXLcS_WV7gACnBwAAsZRxhXGsrj-QloNWyAE'
+    PARTY_GIRL = 'CAACAgUAAxkBAAECkXFg7GNN3M6hwcJb21P74JELHq0mHQACpxwAAsZRxhXfhZz2U_6oByAE'
+    LOLI = 'CAACAgUAAxkBAAECjRVg5uYDQV50eYErYbW-52OzbJ9eWgACkh4AAsZRxhWMjlg7SAq03yAE'
+    NAME_TO_STICKER = {name: Sticker(name, name, 512, 512, False) for name in
+                       [SAD_ZOE, MEGUMIN_THUMB_UP, PARTY_GIRL, LOLI]}
+
+    def __init__(self, token, set_api, log_url, whitelist, db_url="mongodb://datacaker:27017"):
+        self.db = Database(db_url)
+        self.token = token
+        self.set_api = set_api
+        self.log_url = log_url
+        self.whitelist = whitelist
+        self.WAITING_API = 0
+
+    def authorize_and_dispatch(self, update: Update, context: CallbackContext, dispatcher, whitelist):
```

Comment:
Whitelist is already given on object init. Remember to change pytest accordingly